provisioner/aws.go

```diff
@@ -107,6 +108,10 @@ type s3UploaderAPI interface {
 	Upload(input *s3manager.UploadInput, options ...func(*s3manager.Uploader)) (*s3manager.UploadOutput, error)
 }
 
+type kmsAPI interface {
+	Encrypt(input *kms.EncryptInput) (*kms.EncryptOutput, error)
+}
```

Comment:
Please use the `kmsiface`: https://github.com/aws/aws-sdk-go/tree/master/service/kms/kmsiface We have not done it in other places in this project, but I think it makes more sense than defining a custom interface each time.

Here's an example of stsiface being used in a similar way: https://github.com/mikkeloscar/kube-aws-iam-controller/blob/2c39d429695d6ae624d794fa19e7755c2b9a0112/credentials_getter_test.go#L16

Reply:
What magic is this :)